phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/db/SqlitePaymentsDb.kt

```diff
@@ -383,10 +402,32 @@ class SqlitePaymentsDb(
         paymentHash: ByteVector32,
         channelId: ByteVector32
     ) {
+        val database = _doNotFreezeMe.database
         val inQueries = _doNotFreezeMe.inQueries
+        val metaQueries = _doNotFreezeMe.metaQueries
 
         withContext(Dispatchers.Default) {
-            inQueries.updateNewChannelReceivedWithChannelId(paymentHash, channelId)
+            database.transaction {
+                inQueries.updateNewChannelReceivedWithChannelId(paymentHash, channelId)
+                metaQueries.mapChannelId(channelId, paymentHash)
+            }
+        }
+    }
+
+    suspend fun updateNewChannelConfirmed(
+        channelId: ByteVector32,
+        receivedAt: Long
+    ) {
+        val database = _doNotFreezeMe.database
+        val inQueries = _doNotFreezeMe.inQueries
+        val metaQueries = _doNotFreezeMe.metaQueries
+
+        withContext(Dispatchers.Default) {
+            database.transaction {
+                metaQueries.fetchPaymentHash(channelId)?.let { paymentHash ->
+                    inQueries.updateNewChannelConfirmed(paymentHash, receivedAt)
```

Comment:
Yeah, I had the same fear awhile back. This isn't the only place this particular situation happens (transaction within another transaction). But I discovered it's basically a no-op. From SQLDelight:

```kotlin
/**
 * Starts a Transaction and runs body in that transaction.
 * Throws: IllegalStateException - if noEnclosing is true and there is already an active Transaction on this thread.
 */
fun transaction(
    noEnclosing: Boolean = false,
    body: TransactionWithoutReturn.() -> Unit
  )
```

So if there's already a transaction, it doesn't start another transaction. We can also be more explicit about it, like we do here:

https://github.com/ACINQ/phoenix/blob/4427e0c47924c93e05fd044e152017b4e05edac5/phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/db/payments/OutgoingQueries.kt#L61

Although I'm not sure that's any more clear. Because `noEnclosing = false` is kinda confusing. Maybe if it said something like `allowSharedParentTransaction = true` it would be helpful...

Reply:
Ah interesting thanks! The double negation is painful indeed 🥵